### ISSUE TITLE
"not yet implemented" error when trait objects are present

### ISSUE THREAD

**Issue by brendanzab** [joshua-maros/ouroboros#21]
This sample code:

```rust
#[ouroboros::self_referencing]
pub struct Test {
    field0: String,
    field1: Box<dyn std::any::Any>,
    #[borrows(field0)]
    field2: &'this str,
}
```

Results in this error:

```
error: custom attribute panicked
  --> fathom/src/driver.rs:34:1
   |
34 | #[ouroboros::self_referencing]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: message: not implemented
```

**Comment by joshua-maros** [comment]
Found the cause, this is now fixed in `0.9.1`.

**Comment by brendanzab** [comment]
Thanks! 🎉
